### PR TITLE
Major changes on the documentation's formatting

### PR DESCRIPTION
**Pull request by foursixnine**
I updated a bit the documentation's formatting, the main idea behind this is to be able to generate a full document from the index, that can be published or distributed through the open.qa website. 

I kept in most cases the same structure as it was before, so looking individually at the documentation directly on the repo should render in most cases almost the same content as before. 

One thing to note is that not everything is perfect, still a lot of formatting to do (And this is just in the openQA documentation... backend is still in progress)... But a html single page can be generated now, with better navigation, and also a single pdf file.

One idea I have is to move the documentation to os-autoinst/documentation, but I guess we're too early in time for this. 

In future changes (i.e another PR), the +api+ section would be replaced by something generated by the `Pod::AsciiDoctor` module.

Examples of the generated single book is here in [Pdf](https://w3.nue.suse.com/~szarate/openQA/docs/openqa-documentation-20161111.pdf) and [html](https://w3.nue.suse.com/~szarate/openQA/docs/openqa-documentation-20161111.html)

### CODE REVIEW

**Comment by foursixnine** [comment]
I tried to solve the conflict (been having a lot of fun with git rebase), but I have played too much with git for the day lol...

File: docs/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+:toc: left
+:toclevels: 6
+:author: openSUSE Team at SUSE
```
Comment (coolo) [inline docs/index.asciidoc]: as this team no longer exist, just make it SUSE

Comment (coolo) [inline docs/index.asciidoc]: or más misterioso: openQA Team :)

**Comment by okurz** [comment]
Please rebase and get rid of the merge commit. If you ask I will help with git. And please also split in smaller commits and do it step wise.

**Comment by foursixnine** [comment]
I just added the output of the generation on the description, but in any case: [pdf version](https://w3.nue.suse.com/~szarate/openQA/docs/openqa-documentation-20161111.pdf) and [html version](https://w3.nue.suse.com/~szarate/openQA/docs/openqa-documentation-20161111.html)

**Comment by foursixnine** [comment]
@okurz I got rid of the merge message. I decided to keep the last commit message mostly due to the fact that it's still the same formatting through the documents. 

I will try to make next commits on documentation more atomic :)

**Comment by aaannz** [comment]
I super like I can now easily generate complete pdf. :+1:

File: docs/Installing.asciidoc [hunk 2]
```diff
@@ -52,35 +49,31 @@ zypper in openQA
...
 # openSUSE Only
 # You can check what modules are enabled by using 'a2enmod -l'
-a2enmod headers
-a2enmod proxy
-a2enmod proxy_http
-a2enmod proxy_wstunnel
+$ a2enmod headers
+$ a2enmod proxy
+$ a2enmod proxy_http
+$ a2enmod proxy_wstunnel
```
Comment (okurz) [inline docs/Installing.asciidoc]: doing this will make it more obvious which commands have to be entered but then copy-paste of the commands does not work anymore so I object

File: docs/Installing.asciidoc [hunk 3]
```diff
@@ -113,18 +106,19 @@ systemctl restart httpd
 --------------------------------------------------------------------------------
 
 The openQA web UI should be available on http://localhost/ now. To ensure
-openQA runs on each boot, you should also +systemctl enable+ the same services.
+openQA runs on each boot, you should also +systemctl enable+ the same services
+including apache.
 
 [source,sh]
 --------------------------------------------------------------------------------
+systemctl enable apache2
```
Comment (okurz) [inline docs/Installing.asciidoc]: that's an unrelated change and should be in its own commit.

File: docs/Installing.asciidoc [hunk 4]
```diff
@@ -173,13 +167,15 @@ installed 'os-autoinst' from packages make sure to pass +--isotovideo+ option
 to point to the checkout dir where isotovideo is, not to +/usr/lib+! Otherwise
 it will have trouble finding its perl modules.
 
-Advanced configuration
-----------------------
-[id="advanced"]
+== Where to now?
+
+From this point on, you can refer to the link:GettingStarted.asciidoc#testing-opensuse-or-fedora[getting started] guide to
+fetch the tests cases and possibly take a look at link:WritingTests.asciidoc[Test Developer Guide]
```
Comment (okurz) [inline docs/Installing.asciidoc]: shouldn't it say "take a look at *the* Test Developer Guide"?

File: docs/Pitfalls.asciidoc [hunk 5]
```diff
@@ -1,16 +1,35 @@
...
+user can share the environment without problems.
+
+This approach will lead to a problem when the openqa package is updated, since the
+directory permissions will be changed again, nothing a chmod or chgrp can't fix.
+
+[source,bash]
+----
+$ chmod -R g+rwx /var/lib/openqa
+$ chgrp -R openqa /var/lib/openqa
+----
```
Comment (okurz) [inline docs/Pitfalls.asciidoc]: that's an unrelated change and I suggest to keep it around for later, first let's get the restructuring finished.

File: docs/WritingTests.asciidoc [hunk 6]
```diff
@@ -184,14 +180,15 @@ on GitHub] for examples.
...
+== Test Development tricks
+
+=== Modifying setting of an existing test
 
 There is no interface to modify existing tests but the clone script
 can be used to create a new job that adds, removes or changes
 settings:
 
-[source,sh]
+[source,bash]
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: what's wrong with `sh`?

File: docs/WritingTests.asciidoc [hunk 7]
```diff
@@ -506,10 +505,13 @@ sub run {
     # continue with the test
 }
 --------------------------------------------------------------------------------
+====
 
-Getting info about parents / children
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Getting information about parents parents and childrens
```
Comment (okurz) [inline docs/WritingTests.asciidoc]: double "parents"

File: docs/WritingTests.asciidoc (hunk 6 above)
Comment (okurz) [inline docs/WritingTests.asciidoc]: ok, it's just a syntax highlighting marker but I always prefer "sh" if one does not actually need bash. `sh` could be more lightweight, i.e. your system starts up faster if you don't need bash ;-)

**Comment by foursixnine** [comment]
aaaand we're back to the need for a rebase... :)

File: docs/Pitfalls.asciidoc (hunk 5 above)
Comment (foursixnine) [inline docs/Pitfalls.asciidoc]: @okurz Fixed :) it is now at https://github.com/os-autoinst/openQA/pull/997 Just the needle caching.

**Comment by okurz** [comment]
I don't know how you managed to do this commit mess. Try `git rebase origin/master`

**Comment by foursixnine** [comment]
@okurz fixed the mess :) Some times the ballmer peak helps. Added the changes you suggesed. also this depends on another pr (I think that's why there are two commits on this one)

**Comment by foursixnine** [comment]
Aha! Managed to do a rebase without screwing up this time! (Now that the other PR #997 is merged... i just need this one...)

**Comment by aaannz** [comment]
:+1:  any further comments @okurz ?
